src/main/java/com/linecorp/armeria/server/VirtualHostBuilder.java

```diff
@@ -272,4 +299,8 @@ public VirtualHost build() {
     public String toString() {
         return VirtualHost.toString(getClass(), defaultHostname, hostnamePattern, sslContext, services);
     }
+
+    public ServerBuilder and() {
+        return serverBuilder;
```

Comment:
How about spliting `VirtualHostBuilder` into three classes?

- `AbstractVirtualHostBuilder` without `and()` and `build()`
- `VirtualHostBuilder` with `build()` only
- `ChainedVirtualHostBuilder` with `and()` only

Then, `ServerBuilder.with*VirtualHost()` could return `ChainedVirtualHostBuilder`.